pkgs/development/interpreters/python/hooks/python-imports-check-hook.sh

```diff
@@ -1,13 +1,35 @@
 # Setup hook for checking whether Python imports succeed
 echo "Sourcing python-imports-check-hook.sh"
 
+_doPythonImportsChecks() (
+    imports="$1"
+    cd $out
+    @pythonCheckInterpreter@ -c 'import sys; import importlib; [importlib.import_module(mod) for mod in sys.argv[1:]]' $imports
+)
+
 pythonImportsCheckPhase () {
     echo "Executing pythonImportsCheckPhase"
 
     if [ -n "$pythonImportsCheck" ]; then
         echo "Check whether the following modules can be imported: $pythonImportsCheck"
+        # FIXME: should this be removed? See #201277
         export PYTHONPATH="$out/@pythonSitePackages@:$PYTHONPATH"
-        ( cd $out && eval "@pythonCheckInterpreter@ -c 'import os; import importlib; list(map(lambda mod: importlib.import_module(mod), os.environ[\"pythonImportsCheck\"].split()))'" )
+        ( # FIXME: remove this debugging helper before merge
+            set +e
+            echo "pythonImportsCheckPhase PATH and PYTHONPATH diff:"
+            buildPythonPath "$out $propagatedBuildInputs $propagatedNativeBuildInputs"
+            diff --unified --color=always \
+                <(echo "$PATH"               | sed -e 's/:/\n/g' | sort --unique)  \
+                <(echo "$program_PATH"       | sed -e 's/:/\n/g' | sort --unique) | tail -n+4
+            diff --unified --color=always \
+                <(echo "$PYTHONPATH"         | sed -e 's/:/\n/g' | sort --unique)  \
+                <(echo "$program_PYTHONPATH" | sed -e 's/:/\n/g' | sort --unique) | tail -n+4
+            true
+        )
+        (
+            buildPythonPath "$out $propagatedBuildInputs $propagatedNativeBuildInputs"
```

Comment:
Is there a need to pass propagated inputs as well? $out should already have nix/propagated-build-inputs.

Reply:
Oh, I should look into that